### PR TITLE
doc(migration): Add migration docs

### LINKED ISSUE #463: Add original helm -> helm classic (-> k8s/helm) migration doc
Some brief explanation of the helm binary being renamed in "Helm Classic" and default home directory _moving_, both for the sake of clearing the way for k8s/helm.

Migration command might be something like:

```
$ # First set HELMC_HOME as desired, IF desired
$ mv $(helm home) $(helmc home) && rm $(which helm)
```

### PR DESCRIPTION
**Pull request by krancour**
Fixes #463 

Explains the original Helm -> Helm Classic rebranding and steps for users of the original Helm to follow to complete the migration.

### CODE REVIEW

**Comment by krancour** [comment]
cc @kmala @slack

File: README.md [hunk 1]
```diff
@@ -37,6 +37,34 @@ curl -s https://get.helm.sh | bash
   - the CI build pipeline: [ ![Download](https://api.bintray.com/packages/deis/helm-ci/helm-classic/images/download.svg) ](https://bintray.com/deis/helm-ci/helm-classic/_latestVersion#files)
 2. Unzip the package and make sure `helmc` is available on the PATH.
 
+### Migration Notes
+
+If you are a user of the original Helm tool (versions prior to v0.7.0), take note that Helm Classic is a _re-branding_ of that tool-- new name, same great taste!
+
+__Helm Classic is fully compatible with existing Helm charts!__
```
Comment (arschles) [inline README.md]: maybe make a clarifying note here that charts designed for `kubernetes/helm` won't be compatible? someone reading this in "the future" may interpret this sentence differently than they do now

File: README.md [hunk 2]
```diff
@@ -37,6 +37,34 @@ curl -s https://get.helm.sh | bash
...
+
+Anyone upgrading to Helm Classic from an older version should complete the following steps to fully replace their existing tool with Helm Classic. Doing so will clear the path for eventual installation of the new and improved Helm ([kubernetes/helm][k8s-helm]).
```
Comment (arschles) [inline README.md]: "upgrading to Helm Classic" sounds a little weird - maybe "switching to Helm Classic"?

**Comment by arschles** [comment]
made [two](https://github.com/helm/helm-classic/pull/472#discussion_r62368871) [suggestions](https://github.com/helm/helm-classic/pull/472#discussion_r62368920) above but I don't think either are blocking for this PR. I'm good with this

File: README.md (hunk 1 above)
Comment (krancour) [inline README.md]: Maybe if I say "_previously_ existing Helm charts?"

File: README.md (hunk 2 above)
Comment (krancour) [inline README.md]: Agreed.  It does sound odd.  Let's go with "migrating."

File: README.md (hunk 1 above)
Comment (arschles) [inline README.md]: `previously` sgtm :+1:

File: README.md (hunk 2 above)
Comment (arschles) [inline README.md]: yep, `migrating` sgtm :+1:

**Comment by krancour** [comment]
@arschles fyi, updated as we discussed.

File: docs/index.md [hunk 3]
```diff
@@ -9,6 +9,48 @@
...
+From a Linux or Mac OS X client:
+```
+curl -s https://get.helm.sh | bash
+```
+
+*or*:
+
+1. Grab a prebuilt binary from:
+  - the latest release: [ ![Download](https://api.bintray.com/packages/deis/helm/helm-classic/images/download.svg) ](https://bintray.com/deis/helm/helm-classic/_latestVersion#files)
+  - the CI build pipeline: [ ![Download](https://api.bintray.com/packages/deis/helm-ci/helm-classic/images/download.svg) ](https://bintray.com/deis/helm-ci/helm-classic/_latestVersion#files)
```
Comment (kmala) [inline docs/index.md]: the bintray links should have /helmc/_latestVersion instead of helm-classic/_latestVersion because our packages are named so

Comment (krancour) [inline docs/index.md]: Good catch!

**Pull request merged**